coding-standards/html-css/README.md

```diff
@@ -101,68 +154,45 @@ Using the extend function in Sass, PostCSS or other tools quickly leads to bloat
 
 Do not qualify selectors unless you have a compelling reason to do so. If `.nav {}` will work, do not use `ul.nav {};`. To do so would limit the places you can use the .nav class and increase the specificity of the selector.
 
-If you have to increase specificity, double the class name. For example: `.button.button` is twice as specific as `.button`. When resorting to this, document why it's necessary.
-
 
 ### JavaScript hooks
 
-Never couple JavaScript and CSS. If you need a class as an anchor for your javascript, prepend it with `.js` or use a data attribute.
+Never couple JavaScript and CSS. If you need a class as an anchor for your javascript, prepend it with `.js-` or use a data attribute.
 
 ```html
 <a href="img.jpg" class="js-toggle-lightbox">View large image</a>
 
 <a href="img.jpg" data-handler="lightbox">View large image</a>
+
+<div class="lightbox js-lightbox"></div>
+
+<div class="lightbox" data-enhancer="lightbox"></div>
 ````
 
 
 ### Modularity
 
 If you have a module called `.event`, there should be a corresponding `event.css` (or Sass equivalent) file. Preferably, there should also be an HTML file to match.
 
-In the file `event.css`, all selectors should start with `.event`. Using other module selectors makes the modules dependant on each other and no longer modular.
-
-The one exception to this rule is selectors for [state classes](#states), which can be used throughout your application.
-
-```css
-/* bad */
-.block-list {
-  padding: 20px; background: #ddd;
-}
-
-.block-list .event {
-  margin-bottom: 30px;
-}
-```
-
-```css
-/* good */
-.block-list {
-  padding: 20px; background: #ddd;
-}
-
-.block-list__event {
-  margin-bottom: 30px;
-}
-```
-
-
-If you find that there is a part of a module which can be independently re-used, refactor it to its own module. The more styles put together in a module, the harder to maintain it becomes.
-
-When in doubt, split up the module.
```

Comment:
I'm curious about the motivation to remove this section? 
I've always thought this was one of the strongest points of our setup.

Don't get me wrong, especially as someone not working on CSS on a daily basis I wholly accept your changes, but I'm just wondering what prompted this to grow my understanding. 🙂

Reply:
Same

Reply:
I think we're more modular than ever, so don't get me wrong. I do however strongly disagree with how this was probably meant, but it can be interpreted in several ways. So I removed it because it can send the wrong signal, though I specifically asked @MicheleNL to discuss this with me, since we could always argue and/or come up with a better example.

So what I really disagree with is the following scenario — and I think this is how it was written / meant to be, especially looking at a project like Entrée, which was horrible to refactor due to these practices:

```html
/* parent */
<section class="block-list"></section>

/* potential child */
<div class="event"></div>
```

So what I really disagree with is this solution, which is where you're still actually 'mixing' modules in your templates, partials, arguments, etc...:

```html
/* bad */
<section class="block-list">
    <div class="block-list__event event"></div>
</section>
```

I personally do NOT care about an extra 'unnecessary' DOM-element, because I think keeping everything scoped is always better than juggling with classes. 

To me `<div class="block-list__event event"></div>` already start to feel like utility-class-territory. You will have to look up both `.block-list` and `.event`, and you're never sure at first glance if the thing is more of an `.event`-module or more of a `.block-list__event`-module. It makes it harder for unfamiliar developers, and I believe it's more error-prone and actually less 'modular'.

So what I personally use / would favour...

```html
/* good */
<section class="block-list">
    <div class="block-list__event">
         /* @import the module */
    </div>
</section>

/* maybe even better, looking at the class */
<ul class="block-list">
    <li>
         /* @import the .event module */
    </li>
    <li>
         /* @import the .event module */
    </li>
</ul>
```

... and then for styling:

```css
.block-list {
    @include bare-list;
}
.block-list > li + li {
    margin-top: 30px;
}


..event {
    /* some fancy styling which doesn't care about `.block-list`, or wherever it's used */
}
```

That's just my 2 cents. So the reason I've removed it for now is to start a discussion, and because I think due to new insights the practice is obsolete. Just like when we though using utility classes and extends for everything was a good idea 👍

Reply:
I whole-heartedly agree with the "good" version. 👍 
In my mind this is actually how it was meant when it was written, but I agree that it's *not* how it was implemented in practice.

I did notice we're moving towards more element selectors, whereas in olden times we would put classes on everything, and while I am totally fine with element selectors, it's still something we should approach with caution because it makes for really tightly coupled DOM to a certain layout.
I would advice to *never* use a `div` selector for instance because it's way too generic. 
Chances are a bit of content will be added at some point that introduces a second div, or needs a bit of extra nesting. In that case it's far easier to refactor and plug the new content in when a class can be transported to a new element, versus having to touch all the CSS when a bit of DOM is updated.

Reply:
Would you, by the way, in the above example put the `.event` styles inside `event.scss` and `block-list` in `block-list.scss`? 
I do really like that for discoverability. For instance in Schoolwijzer I have no clue whatsoever where a bit of styling is going to be.

Reply:
> I did notice we're moving towards more element selectors, whereas in olden times we would put classes on everything, and while I am totally fine with element selectors, [...]

Not totally true, we've been doing that for years already. This was at the time where Mattijs (and me) chose to simplify stuff. Like making new lines for `.selector:hover` and `.selector:focus`, instead of putting everything SCSS-style inside selectors. We tried to simplify the 'over-BEMming', since creating classes for EVERYTHING made it harder to read. In your CSS and your template. This is of course only justified when the template is simple enough to allow it. But there are no real 'enforced rules' for this. What I mean is...

```html
/* good *
<nav class="site-nav">
    <ul>
        <li><a href="#"></li>
        <li><a href="#"></li>
        <li><a href="#"></li>
    </ul>
</nav>

/* also good */
<nav class="site-nav">
    <ul class="site-nav__list">
        <li class="site-nav__item"><a href="#"></li>
        <li class="site-nav__item"><a href="#"></li>
        <li class="site-nav__item"><a href="#"></li>
    </ul>
</nav>
```

I usually use the former version, but if someone uses the latter one, I wouldn't care. I did so too, before. In this case the module is that simple, that I'm totally fine with tag selectors, and in your CSS it's actually clear that's it a `li`, and not some `div`. It makes the DOM more readable in this case, both in the template and CSS. 

Anyhow: it's only meant for small/simple modules, and usually only for stuff that will be there once, like `nav`, `ul`, `li`, `dl`, `dt`, `dd`, `details`, etc. 

> it's still something we should approach with caution because it makes for really tightly coupled DOM to a certain layout.

It depends. I wouldn't mind if CSS breaks, since adding extra elements in between existing elements, and relying on classes to 'just keep working'... not sure? But still, we only use tag selectors when appropriate, see the above tags. So it would probably break regardless of class/tag selector.

> I would advice to never use a div selector for instance because it's way too generic.
Chances are a bit of content will be added at some point that introduces a second div, or needs a bit of extra nesting. In that case it's far easier to refactor and plug the new content in when a class can be transported to a new element, versus having to touch all the CSS when a bit of DOM is updated.

As far as I know we never use `div` selectors (anymore), so I strongly agree 👍

Reply:
Haha, I think we're in agreement. 😄 
Let's merge this!

Reply:
Actually, the story about tag selectors was already more or less in there, but hidden in the examples of 'Nesting': https://github.com/grrr-amsterdam/guidelines/tree/master/coding-standards/html-css#nesting 

It also specifically said this was a good idea...

```html
  <div class="glass">
    <div class="glass__contents beer">
    ...
    </div>
  </div>
```

... which is still coupling modules, so I've simplified that example to avoid any confusion. Furthermore:

- I've re-added the examples we were discussing here, but provided more examples/context: https://github.com/grrr-amsterdam/guidelines/blob/1621363bb609cb7e0218dc51bef479bed6861b72/coding-standards/html-css/README.md#modularity
- I did add a small section about tag selectors as well: https://github.com/grrr-amsterdam/guidelines/blob/1621363bb609cb7e0218dc51bef479bed6861b72/coding-standards/html-css/README.md#tag-selectors